app/src/main/java/com/example/weatherapp/viewmodel/factories/DetailsViewModelFactory.kt

```diff
@@ -0,0 +1,12 @@
+package com.example.weatherapp.viewmodel.factories
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.weatherapp.viewmodel.WeatherDetailsViewModel
+
+class DetailsViewModelFactory(private val city: String): ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return WeatherDetailsViewModel(SavedStateHandle(), city) as T
```

Comment:
Have you checked the different ways the Android framework provides SavedStateHandle to the ViewModel?

I highly recommend you reading through the docs I posted earlier as those are very useful and have plenty of good stuff in there:

https://developer.android.com/topic/libraries/architecture/viewmodel/viewmodel-savedstate

Reply:
This class has been since removed but I have also read the documentation sent to me